bootcompiler/src/main/scala/org/mozartoz/bootcompiler/transform/OnStackMarking.scala

```diff
@@ -0,0 +1,245 @@
+package org.mozartoz.bootcompiler
+package transform
+
+import ast._
+import oz._
+import symtab._
+import scala.collection._
+
+/**
+ * Declared is the state of a variable that has been declared by a local node but that
+ *   has not been encountered yet. If at the end of a local node, a variable is still in that state,
+ *   it is useless and can be "put on the stack".
+ *
+ * A variable is Captured when it has been captured by a procedure from its declaration until here,
+ *   but that has not been encountered in any other way. When a call node will be reached, every
+ *   captured variable will be set to "ToInstantiate" in order to avoid hazards of having a link to it
+ *   in the called procedure. There is however a hope for seing it become BoundFirst before a call happens.
+ *
+ * BoundFirst is the state of variable that can be put on the stack whatever happens in the next instructions.
```

Comment:
Please describe how BoundFirst can happen. I know the name is almost self-explaining but still :)